bot/src/discord_commands/schedule.js

```diff
@@ -0,0 +1,165 @@
+      intervals.splice(index, 1);
+      msg.channel.createMessage('Scheduled command stopped succesfully.');
+    } else {
+      msg.channel.createMessage('Error: Scheduled command with ID ' + suffixArray[1] + ' not found.');
+    }
+  } else if (suffixArray.length !== 3) {
+    msg.channel.createMessage('Error: You must specify frequency in hours (or use \'daily\' or \'weekly\'), and a starting time (or use \'now\'). Frequency formats consist of number and unit. Valid units are \'s\', \'m\', \'h\', \'d\', \'w\'. Starting time valid fotmat is \'hh:mm\'.');
+  } else {
+    switch (suffixArray[0]) {
+      case 'daily':
+        freq = 24 * 60 * 60 * 1000;
+        freqType = "d";
+        break;
+      case 'weekly':
+        freq = 7 * 24 * 60 * 60 * 1000;
+        freqType = "w";
+        break;
+      default:
+        let regex = new RegExp(/[0-9]+[smhdw]/, 'i');
+        if (regex.test(suffixArray[0])) {
+          let index = suffixArray[0].length - 1;
+          let freqValue = Number.parseInt(suffixArray[0].substring(0, index));
+          freqType = suffixArray[0].substring(index);
+          switch (freqType) {
+            case 's':
+              freq = freqValue * 1000;
+              break;
+            case 'm':
+              freq = freqValue * 1000 * 60;
+              break;
+            case 'h':
+              freq = freqValue * 1000 * 60 * 60;
+              break;
+            case 'd':
+              freq = freqValue * 1000 * 60 * 60 * 24;
+              break;
+            case 'w':
+              freq = freqValue * 1000 * 60 * 60 * 24 * 7;
+              break;
+          }
+        } else {
+          msg.channel.createMessage('Error: You must specify frequency. Frequency formats consist of number and unit. Valid units are \'s\', \'m\', \'h\', \'d\', \'w\'.');
+          return;
+        }
+        break;
+    }
+    switch(suffixArray[1]) {
+      case 'now':
+        start = new Date();
+        start.setSeconds(0);
+        start.setMilliseconds(0);
+        break;
+      default:
+        let regex = new RegExp(/[0-9]{1,2}:[0-9]{2}/, 'i');
+        if (regex.test(suffixArray[1])) {
+          start = new Date();
+          let hours = suffixArray[1].split(':')[0];
+          let minutes = suffixArray[1].split(':')[1];
+          start.setHours(hours);
+          start.setMinutes(minutes);
+          start.setSeconds(0);
+          start.setMilliseconds(0);
+        } else {
+          msg.channel.createMessage('Error: Start time must be in format hh:mm.');
+          return;
+        }
+        break;
+    }
+    level = suffixArray[2];
+    let id = intervals.length > 0 ? intervals[intervals.length-1].id + 1 : 0;
+    intervals.push({
+      id: id,
+      channel: msg.channel.name,
+      frequency: suffixArray[0],
+      start: start.toString(),
+      level: level,
+      interval: setTimeout(() => {
+        setTimer(id, msg, level, monochrome, freq, start);
+      }, 0)
```

Comment:
Why setTimeout with a timeout of `0`?

Reply:
So it pushes the element **before** it searches for its index inside the setTimer function. I've just seen that there's another function that accomplishes the same thing and it's slightly faster, setImmediate(), so I'll change it.

Reply:
Gotcha, it's a bit problematic though due to error handling (if setTimer throws an exception here, the whole process will crash). setImmediate() has the same problems.

It doesn't seem like you read the `interval` property in setTimer, so could you just do this?

```js
intervals.push({
      id: id,
      channel: msg.channel.name,
      frequency: suffixArray[0],
      start: start.toString(),
      level: level,
});

setTimer(id, msg, level, monochrome, freq, start);
```

Even if clearTimeout happens somewhere, it's safe to do `clearTimeout(undefined)`. It's just a NOP.

Reply:
You're right. interval is set inside setTimer (so it can be referenced and stopped) but it's not necessary here.